coderd/users.go

```diff
@@ -1015,23 +1029,52 @@ func (api *API) putUserAppearanceSettings(rw http.ResponseWriter, r *http.Reques
 		return
 	}
 
-	updatedSettings, err := api.Database.UpdateUserAppearanceSettings(ctx, database.UpdateUserAppearanceSettingsParams{
+	if !isValidFontName(params.TerminalFont) {
+		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+			Message: "Unsupported font family.",
+		})
+		return
+	}
+
+	updatedThemePreference, err := api.Database.UpdateUserThemePreference(ctx, database.UpdateUserThemePreferenceParams{
 		UserID:          user.ID,
 		ThemePreference: params.ThemePreference,
 	})
 	if err != nil {
 		httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
-			Message: "Internal error updating user.",
+			Message: "Internal error updating user theme preference.",
+			Detail:  err.Error(),
+		})
+		return
+	}
+
+	updatedTerminalFont, err := api.Database.UpdateUserTerminalFont(ctx, database.UpdateUserTerminalFontParams{
+		UserID:       user.ID,
+		TerminalFont: string(params.TerminalFont),
+	})
+	if err != nil {
+		httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+			Message: "Internal error updating user terminal font.",
 			Detail:  err.Error(),
 		})
 		return
 	}
 
 	httpapi.Write(ctx, rw, http.StatusOK, codersdk.UserAppearanceSettings{
-		ThemePreference: updatedSettings.Value,
+		ThemePreference: updatedThemePreference.Value,
+		TerminalFont:    codersdk.TerminalFontName(updatedTerminalFont.Value),
 	})
 }
 
+func isValidFontName(font codersdk.TerminalFontName) bool {
+	switch font {
+	case codersdk.TerminalFontIbmPlexMono, codersdk.TerminalFontFiraCode, "":
```

Comment:
Suggestion: It would be nice if these could be defined in one place, and shared between fe/be. Don't really have a suggestion for how to achieve it, just what I'd like to see so we don't have to touch a lot of places when/if adding a new font.